pkg/webui/components/navigation/side/item/index.js

```diff
@@ -127,69 +138,105 @@ const CollapsableItem = ({
+    }
+  })
+
+  const subItemActive = subItems.some(item => item.path === currentPathName)
+
+  return (
+    <>
+      <button
+        className={classnames(style.button, {
+          [style.buttonActive]: isActive,
+          [style.linkActive]: subItemActive,
         })}
-      />
-    </button>
-    <SideNavigationList isMinimized={isMinimized} depth={depth + 1} isExpanded={isExpanded}>
-      {children}
-    </SideNavigationList>
-  </Fragment>
-)
+        type="button"
+        onClick={onClick}
+      >
+        {icon && <Icon icon={icon} className={style.icon} />}
+        <Message content={title} className={style.message} />
+        <Icon
+          icon="keyboard_arrow_down"
+          className={classnames(style.expandIcon, {
+            [style.expandIconOpen]: isExpanded,
+          })}
+        />
+      </button>
+      <Dropdown className={style.flyOutList} onItemsClick={onDropdownItemsClick}>
+        <Dropdown.HeaderItem title={title.defaultMessage} />
+        {subItems.map(item => (
+          <Dropdown.Item key={item.path} title={item.title} path={item.path} icon={item.icon} />
+        ))}
+      </Dropdown>
+      <SideNavigationList depth={depth + 1} isExpanded={isExpanded} className={style.subItems}>
+        {children}
+      </SideNavigationList>
+    </>
+  )
+}
 
 CollapsableItem.propTypes = {
   children: PropTypes.node,
+  currentPathName: PropTypes.string.isRequired,
   depth: PropTypes.number.isRequired,
   icon: PropTypes.string,
   isActive: PropTypes.bool.isRequired,
   isExpanded: PropTypes.bool.isRequired,
-  isMinimized: PropTypes.bool,
   onClick: PropTypes.func.isRequired,
+  onDropdownItemsClick: PropTypes.func,
   title: PropTypes.message.isRequired,
 }
 
 CollapsableItem.defaultProps = {
   children: undefined,
   icon: undefined,
-  isMinimized: false,
+  onDropdownItemsClick: () => null,
 }
 
-const LinkItem = ({ onClick, title, icon, exact, path }) => (
-  <NavigationLink
-    onClick={onClick}
-    className={style.link}
-    activeClassName={style.linkActive}
-    exact={exact}
-    path={path}
-  >
-    {icon && <Icon icon={icon} className={style.icon} />}
-    <Message content={title} className={style.message} />
-  </NavigationLink>
-)
+const LinkItem = ({ onClick, title, icon, exact, path, onDropdownItemsClick }) => {
+  const handleLinkItemClick = React.useCallback(
+    event => {
+      document.activeElement.blur()
+      onClick(event)
+    },
+    [onClick],
+  )
+
+  return (
+    <>
+      <NavigationLink
+        onClick={handleLinkItemClick}
+        className={style.link}
+        activeClassName={style.linkActive}
+        exact={exact}
+        path={path}
+      >
+        {icon && <Icon icon={icon} className={style.icon} />}
+        <Message content={title} className={style.message} />
+      </NavigationLink>
+      <Dropdown className={style.flyOutList} onItemsClick={onDropdownItemsClick}>
+        <Dropdown.Item title={title} path={path} showActive={false} icon={''} />
```

Comment:
```suggestion
        <Dropdown.Item title={title} path={path} showActive={false} icon={''} tabIndex="-1" />
```

Let's use `tabIndex` -1 here, since otherwise, users have to tab twice per entry. You need to pass this in the dropdown item component.

Reply:
Great point